cpp/libcudf_kafka/CMakeLists.txt

```diff
@@ -13,112 +13,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
-
-project(CUDA_KAFKA VERSION 0.15.0 LANGUAGES C CXX CUDA)
-
-# TODO: Since we have no actual CUDA code in cudf_kafka this should be removed in the future
-# in favor of using FindCUDAToolkit to get the needed CUDA include headers
-if(NOT CMAKE_CUDA_COMPILER)
-  message(SEND_ERROR "CMake cannot locate a CUDA compiler")
-endif(NOT CMAKE_CUDA_COMPILER)
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+project(CUDA_KAFKA VERSION 0.19.0 LANGUAGES C CXX)
 
 ###################################################################################################
-# - build type ------------------------------------------------------------------------------------
+# - Build options
+
+option(BUILD_TESTS "Build tests for libcudf_kafka" ON)
 
-# Set a default build type if none was specified
-set(DEFAULT_BUILD_TYPE "Release")
+message(VERBOSE "CUDF_KAFKA: Build gtests: ${BUILD_TESTS}")
 
-if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-  message(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
-  set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE
-      STRING "Choose the type of build." FORCE)
-  # Set the possible values of build type for cmake-gui
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
-    "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+# Set CUDF_SOURCE_DIR to ensure that cmake Modules from parent cudf have the correct context
+set(CUDF_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../")
 
 ###################################################################################################
 # - compiler options ------------------------------------------------------------------------------
 
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-
-# To apply RUNPATH to transitive dependencies (this is a temporary solution)
-set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--disable-new-dtags")
-set(CMAKE_EXE_LINKER_FLAGS "-Wl,--disable-new-dtags")
-
-# Build options
-option(BUILD_TESTS "Configure CMake to build tests" ON)
+# * find CUDAToolkit package
+# * determine GPU architectures
+# * enable the CMake CUDA language
+# * set other CUDA compilation flags
+include(../cmake/Modules/ConfigureCUDA.cmake)
```

Comment:
I think this is causing problems. There's no device code / actual cuda configuration needed for libcudf, and we're doing this before importing cudf / rmm / etc. which causes the `CUDA` language to be enabled which breaks things I think?

Reply:
In my dummy CMakeLists.txt for https://github.com/rapidsai/cudf/pull/7658, `CUDA` had to be enabled for the `find_package(cudf)` to succeed.

Reply:
Ok, that would make sense actually since I removed CUDA in a previous commit. Both @kkraus14 and I were both under the impression it was not needed since libcudf_kafka doesn't use CUDA explicitly. Are you saying I need to add it back?

Reply:
My thought is that if someone needs to enable `CUDA` before doing `find_package(cudf)`, we have something broken in our cudf cmake config.

Reply:
It looks like `cuftestutil` and `cudf` both have some CUDA information in the export information. That would require downstream consumers to enable CUDA.

- `cuftestutil` is tricky as it is a static library with CUDA sources, so CMake needs to export that when linking downstream it has CUDA dependencies ( so you need to enable the language )
- `cudf` is leaking via compile flags, would could be fixed

Reply:
Both of these should be resolved by having `cudf-config.cmake` have ( `enable_language(CUDA)` ) in the short term while we sort out the above two issues.